scylla-sstable: add scrub operation (#14260)
scylladb/scylladb: docs/operating-scylla/admin-tools/scylla-sstable.rst

Any errors found will be logged with error level to ``stderr``.

scrub
^^^^^

Rewrites the sstable, skipping or fixing corrupt parts. Not all kind of corruptions can be skipped or fixed by scrub.
Currently this is limited to ordering issues on the partition, row or mutation-fragment level, see `sstable content <scylla-sstable-sstable-content_>`_ for more details.

Scrub has several modes:

> * **abort** - abort the scrub as soon as any error is found (recognized or not). This mode is not all the useful (**validate** mode is strictly an improvement over **abort** mode), it is included for completeness' sake.

Reviewer: ```suggestion
* **abort** - abort the scrub as soon as any error is found (recognized or not). This mode is not all that useful (**validate** mode is strictly an improvement over **abort** mode), it is included for completeness' sake.
```

Author: Done in v2.

Any errors found will be logged with error level to ``stderr``.

scrub
^^^^^

Rewrites the sstable, skipping or fixing corrupt parts. Not all kind of corruptions can be skipped or fixed by scrub.

> Currently this is limited to ordering issues on the partition, row or mutation-fragment level, see `sstable content <scylla-sstable-sstable-content_>`_ for more details.

Reviewer 2: ```suggestion
It is limited to ordering issues on the partition, row, or mutation-fragment level. See `sstable content <scylla-sstable-sstable-content_>`_ for more details.
```

Scrub has several modes:

> * **abort** - abort the scrub as soon as any error is found (recognized or not). This mode is not all that useful (**validate** mode is strictly an improvement over **abort** mode), it is included for completeness' sake.

Reviewer 2: ```suggestion
* **abort** - Aborts the scrub as soon as any error is found (recognized or not). This mode is only included for the sake of completeness. We recommend using the **validate** mode so that all errors are reported.
```

Reviewer 2: @denesb I'm not sure if this is what we want to say. You can rephrase it differently - I just think we shouldn't document anything as "not useful".

> * **skip** - skip over any corruptions found, thus omitting them from the output. Note that skip can end up omitting more than was strictly necessary, but it is guaranteed that it will skip all detectable corruptions.

Reviewer 2: ```suggestion
* **skip** - Skips over any corruptions found, thus omitting them from the output. Note that this mode can result in omitting more than is strictly necessary, but it guarantees that all detectable corruptions will be omitted.
```

> * **segregate** - fix partition/row/mutation-fragment out-of-order errors, by segregating the output into as many sstables as required, such that the content of each output sstable is properly ordered.

Reviewer 2: ```suggestion
* **segregate** - Fixes partition/row/mutation-fragment out-of-order errors by segregating the output into as many SStables as required so that the content of each output SStable is properly ordered.
```

> * **validate** - validate the content of the sstable, reporting any corruptions found. Writes no output sstables. This is the same as the `validate operation <scylla-sstable-validate-operation_>`_ and it is recommended that the latter is used instead.

Reviewer 2: ```suggestion
* **validate** - Validates the content of the SStable, reporting any corruptions found. Writes no output SStables. In this mode, scrub has the same outcome as the `validate operation <scylla-sstable-validate-operation_>`_ - and the validate operation is recommended over scrub.
```

> Output sstables are written to the directory specified via ``--output-directory``. They will be written with the ``BIG`` format and the highest supported sstable format, with generations choosen by scylla-sstable. Generations are chosen such

Reviewer 2: ```suggestion
Output SStables are written to the directory specified via ``--output-directory``. They will be written with the ``BIG`` format and the highest supported SStable format, with generations chosen by scylla-sstable. Generations are chosen such
```

> that they are unique between the sstables written by the current scrub.

Reviewer 2: ```suggestion
that they are unique among the SStables written by the current scrub.
```

> The output directory is expected to be empty, if it isn't, scylla-sstable will abort the scrub. This can be overriden by the ``--unsafe-accept-nonempty-output-dir`` command line flag.

Reviewer 2: ```suggestion
The output directory must be empty; otherwise, scylla-sstable will abort scrub. You can allow writing to a non-empty directory by setting the ``--unsafe-accept-nonempty-output-dir`` command line flag.
```

> Note that scrub will be aborted if an sstable cannot be written because its generation clashes with a pre-existing sstable in the directory.

Reviewer 2: ```suggestion
Note that scrub will be aborted if an SStable cannot be written because its generation clashes with a pre-existing SStable in the output directory.
```
